### PR TITLE
Serialize and de-serialize non-keyword schema values correctly. Fixes #52.

### PR DESCRIPTION
**Pull request by ncalexan**
@rnewman, over to you.

I put the test first so that I could verify it failed before the patch.  Will land together to not break (future?) CI :)

### CODE REVIEW

File: test/datomish/db_test.cljc [hunk 1]
```diff
@@ -781,4 +781,34 @@
...
+        (let [db     (d/db conn)
+              schema (d/schema db)]
+          (is (= true (ds/indexing? schema (d/entid db :db/txInstant))))
+          (is (= true (ds/fulltext? schema (d/entid db :test/fulltext))))))
+
+      ;; Close and re-open same DB.
+      (<? (d/<close conn))
+      (let [conn (<? (d/<connect t))]
+        (try
+          (testing "Boolean values in schema are correct after re-opening"
```
Comment (rnewman) [inline test/datomish/db_test.cljc]: Can we also explicitly check here that keyword-typed schema values come back around?

File: src/datomish/db_factory.cljc [hunk 2]
```diff
@@ -45,26 +45,34 @@
...
+    (let [ident-map (into {} (clojure.set/map-invert idents))
+          ref-tag   (sqlite-schema/->tag :db.type/ref)]
       (->>
-        {:select [:ident :attr :value] :from [:schema]}
-        (s/format)
-        (s/all-rows sqlite-connection))
-      (<?)
+        (->>
+          {:select [:ident :attr :value :value_type_tag] :from [:schema]}
+          (s/format)
```
Comment (rnewman) [inline src/datomish/db_factory.cljc]: You might as well do this at read time, or by hand.

```
"SELECT ident, attr, value, value_type_tag FROM schema"
```

File: src/datomish/db_factory.cljc [hunk 3]
```diff
@@ -45,26 +45,34 @@
...
 
-(defn <symbolic-schema [sqlite-connection]
+(defn <symbolic-schema [sqlite-connection idents]
   "Read the schema map materialized view from the given SQLite store.
   Returns a map (keyword ident) -> (map (keyword attribute -> keyword value)), like
   {:db/ident {:db/cardinality :db.cardinality/one}}."
 
   (go-pair
-    (->>
+    (let [ident-map (into {} (clojure.set/map-invert idents))
```
Comment (rnewman) [inline src/datomish/db_factory.cljc]: You don't need `into` — `map-invert` returns a map.

File: src/datomish/db_factory.cljc [hunk 4]
```diff
@@ -45,26 +45,34 @@
...
+        (<?)
 
-      (group-by (comp (partial sqlite-schema/<-SQLite :db.type/keyword) :ident))
-      (map (fn [[ident rows]]
-             [ident
-              (into {} (map (fn [row]
-                              [(sqlite-schema/<-SQLite :db.type/keyword (:attr row))
-                               (sqlite-schema/<-SQLite :db.type/keyword (:value row))]) rows))])) ;; TODO: this is wrong, it doesn't handle true.
-      (into {}))))
+        (group-by (comp (partial sqlite-schema/<-SQLite :db.type/keyword) :ident))
```
Comment (rnewman) [inline src/datomish/db_factory.cljc]: Pull out the `partial` part and reuse here and on line 73.

**Comment by rnewman** [comment]
LGTM, @ncalexan.

File: test/datomish/db_test.cljc (hunk 1 above)
Comment (ncalexan) [inline test/datomish/db_test.cljc]: Done, keyword and string (:db/doc) values.

File: src/datomish/db_factory.cljc (hunk 2 above)
Comment (ncalexan) [inline src/datomish/db_factory.cljc]: Done throughout this file.

File: src/datomish/db_factory.cljc (hunk 3 above)
Comment (ncalexan) [inline src/datomish/db_factory.cljc]: Thanks, removed a step and didn't clean up.

File: src/datomish/db_factory.cljc (hunk 4 above)
Comment (ncalexan) [inline src/datomish/db_factory.cljc]: Done.

**Comment by ncalexan** [comment]
Landed.

**Pull request closed without merging**